project-base/app/tests/FrontendApiBundle/Unit/Complaint/ComplaintApiFacadeTest.php

```diff
@@ -0,0 +1,215 @@
+        $this->complaintApiFacade->createFromComplaintInputArgument($argument);
+    }
+
+    public function testCreateFromComplaintInputArgumentForCustomerSuccess(): void
+    {
+        $argument = self::getCreateFromComplaintInputArgument();
+
+        $order = $this->createMock(Order::class);
+        $customerUser = $this->createMock(CustomerUser::class);
+        $orderCustomerUser = $this->createMock(CustomerUser::class);
+
+        $complaintData = $this->createMock(ComplaintData::class);
+        $complaint = $this->createMock(Complaint::class);
+
+        $this->security->method('isGranted')->with(CustomerUserRole::ROLE_API_ALL)->willReturn(true);
+
+        $customerUser->method('getCustomer')->willReturn($orderCustomerUser);
+
+        $this->orderApiFacade->method('getByUuid')->with('order-uuid')->willReturn($order);
+        $this->currentCustomerUser->method('findCurrentCustomerUser')->willReturn($customerUser);
+
+        $order->method('getCustomerUser')->willReturn($orderCustomerUser);
+        $order->method('getCustomer')->willReturn($orderCustomerUser);
+
+        $this->complaintNumberSequenceRepository->method('getNextNumber')->willReturn('123');
+        $orderItemMock = $this->createMock(OrderItem::class);
+        $orderItemMock->method('getOrder')->willReturn($order);
+        $orderItemMock->method('getQuantity')->willReturn(1);
+
+        $this->orderItemApiFacade->method('findMappedByUuid')->with(['item-uuid'])
+            ->willReturn(['item-uuid' => $orderItemMock]);
+
+        $this->complaintDataApiFactory->method('createFromComplaintInputArgument')
+            ->willReturn($complaintData);
+        $this->complaintFactory->method('create')->willReturn($complaint);
+
+        $this->em->expects($this->once())->method('persist');
+        $this->em->expects($this->once())->method('flush');
+
+        $result = $this->complaintApiFacade->createFromComplaintInputArgument($argument);
+        $this->assertSame($complaint, $result);
+    }
+
+    public function testCreateFromComplaintInputArgumentForCustomerInvalidAccessUserError(): void
+    {
+        $argument = self::getCreateFromComplaintInputArgument();
+
+        $order = $this->createMock(Order::class);
+        $customerUser = $this->createMock(CustomerUser::class);
+        $customerUser2 = $this->createMock(CustomerUser::class);
+        $orderCustomerUser = $this->createMock(CustomerUser::class);
+
+        $this->security->method('isGranted')->with(CustomerUserRole::ROLE_API_ALL)->willReturn(true);
+
+        $customerUser->method('getCustomer')->willReturn($customerUser2);
+
+        $this->orderApiFacade->method('getByUuid')->with('order-uuid')->willReturn($order);
+        $this->currentCustomerUser->method('findCurrentCustomerUser')->willReturn($customerUser);
+
+        $order->method('getCustomerUser')->willReturn($customerUser2);
+        $order->method('getCustomer')->willReturn($orderCustomerUser);
```

Comment:
this should return a mock of the `Customer` class, to be precise (a different instance than the one used on the line 189)

Reply:
fixed [fixup! fixup! complaints b2b customers support](https://github.com/shopsys/shopsys/pull/3295/commits/cf1dc4153034074866c7924fd034d3dbf959492b)